src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -264,7 +267,7 @@ public virtual void WriteTo(TextWriter writer, string value)
         {
             if (!string.IsNullOrEmpty(value))
             {
-                writer.Write(WebUtility.HtmlEncode(value));
+                writer.Write(HtmlEncoder.HtmlEncode(value));
```

Comment:
Use `HtmlEncoder.HtmlEncode(value, writer)` instead. That way the encoder writes directly to the backing TextWriter.